tests/testStatsdClient.cpp

```diff
@@ -53,43 +53,24 @@ void testErrorConditions() {
     throwOnError(client, false, "Should not be able to resolve a ridiculous ip");
 }
 
-void testReconfigure() {
-    StatsdServer server;
-    throwOnError(server);
-
-    StatsdClient client("localhost", 8125, "first.");
-    client.increment("foo");
-    throwOnWrongMessage(server, "first.foo:1|c");
-
-    client.setConfig("localhost", 8125, "second");
-    client.increment("bar");
-    throwOnWrongMessage(server, "second.bar:1|c");
-
-    client.setConfig("localhost", 8125, "");
-    client.increment("third.baz");
-    throwOnWrongMessage(server, "third.baz:1|c");
-
-    client.increment("");
-    throwOnWrongMessage(server, ":1|c");
-
-    // TODO: test what happens with the batching after resolving the question about incomplete
-    //  batches being dropped vs sent on reconfiguring
-}
-
 void testSendRecv(uint64_t batchSize, uint64_t sendInterval) {
     StatsdServer mock_server;
     std::vector<std::string> messages, expected;
     std::thread server(mock, std::ref(mock_server), std::ref(messages));
 
+    std::mt19937 twister(std::random_device{}());
+    std::uniform_real_distribution<float> dist(0.f, 1.f);
+    auto rand = [&]()->float{return dist(twister);};
```

Comment:
we pass our own custom sampling function so we can fiddle with it to get it to give the results we want